File: Selenium/Activities/Activity22.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.Set;

public class Activity22 {
    public static void main(String[] args) throws InterruptedException {

        // Setup the Webdriver
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver= new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofMillis(1000));

        // Launch the URL and get the title
        driver.get("https://v1.training-support.net/selenium/popups");
        String pageTitle = driver.getTitle();
        System.out.println(" Title of the Page is : " + pageTitle);

        // Locate the Sign In Button
        WebElement signInButton = driver.findElement(By.cssSelector("button.ui.huge.inverted.orange.button"));

        // Build Actions to hover over the sign in button
        Actions builder = new Actions(driver);
        builder.moveToElement(signInButton).build().perform();

        // Print the tool tip Message
        System.out.println(signInButton.getAttribute("data-tooltip"));

        // CLick the SignIn Button
        signInButton.click();

        // Location the username and password Text Box and click Submit
        WebElement userTextBox = driver.findElement(By.id("username"));
        WebElement passTextBox = driver.findElement(By.id("password"));
        userTextBox.sendKeys("admin");
        passTextBox.sendKeys("password");

        WebElement loginInButton = driver.findElement(By.xpath("//button[text()='Log in']"));
        loginInButton.click();

        // Locate the Login Confirmation Message
        WebElement confirmationMessage = driver.findElement(By.id("action-confirmation"));
        System.out.println("Login Confirmation Message is : "+ confirmationMessage.getText());

        // Wait for 3 Seconds and close the browser
        Thread.sleep(3000);
        driver.quit();

    }
}

File: Selenium/Activities/Activity22.py
# Import required packages from selenium
import time

from actions import Actions
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/popups")
    home_title = driver.title
    print("Title of the Home Page is : {}".format(home_title))

    # Locate the Sign In Button
    signInButton = driver.find_element(By.CSS_SELECTOR, "button.ui.huge.inverted.orange.button")

    # Build Actions to hover over the sign in button
    builder = ActionChains(driver)
    builder.move_to_element(signInButton).perform()

    # Print the tool tip Message
    print(signInButton.get_attribute("data-tooltip"))

    # CLick the SignIn Button
    signInButton.click()

    # Location the username and password Text Box and click Submit
    userTextBox = driver.find_element(By.ID, "username")
    passTextBox = driver.find_element(By.ID, "password")
    userTextBox.send_keys("admin")
    passTextBox.send_keys("password")

    loginInButton = driver.find_element(By.XPATH, "//button[text()='Log in']")
    loginInButton.click()

    # Locate the Login Confirmation Message
    confirmationMessage = driver.find_element(By.ID, "action-confirmation")
    print("Login Confirmation Message is : "+ confirmationMessage.text)


    # Wait for 3 Seconds and Close the Browser
    time.sleep(3)
    driver.close()
